cli/collection_describe.go

```diff
@@ -16,6 +16,11 @@ import (
 	"github.com/sourcenetwork/defradb/client"
 )
 
+type CollectionDefinition struct {
```

Comment:
question: does it need to be a public definition of `cli` package?

Seems like it doesn't belong here

Reply:
It is the public output type of the command.  And this is the only place that outputs it.

Where were you thinking it should be?

Reply:
The struct doesn't need to be public. Only its  fields.

Reply:
It's specific to this CLI command and not used anywhere else. I really think this struct shouldn't be public. If you absolutely want it to be public, please at some documentation to the type.

Reply:
When defining a http endpoint would you also declare their return types as private, hiding them from the godocs and any packages embedding the library?

Reply:
Unless that type is returned by a public Go function, there is no use in making the type public. It adds no value to GoDocs as it's not relevant to any of the public functions. 

The HTTP API has to be documented in a specific API doc and the output JSON struct should be shared in that location.

Reply:
:smile: That does surprise me :)

I do see these as the declaration of the data type/shape that is returned from the public endpoints, and as such they should be public.  Is just that the transmission of that shape is via []byte of json instead of []byte of the values only (which is what an instance of a struct really is, only masked/wrapped by the runtime).

I'll ping @nasdf to get his thoughts on this too, as we should be consistent with whatever we do here and the other non-embedded interfaces (including arguably LensVM).

Reply:
It is returned from a public endpoint but that public endpoint isn't a public Go function. That endpoint won't show up in the godocs and the function itself is returning an error. The type/shape should be documented elsewhere but not with a public Go type.

Reply:
My preference would be to define it in the `client` package and reference it in the `cli` and `http`. That would also get rid of the duplicate definition in the `http` client.

Reply:
I would be happy with that option :)

Reply:
Nice okay - I'll try and figure something out.  Some other stuff might get re-jigged slightly as we already have a CollectionDefinition (interface) in client, but hopefully everything will be the better for it.

Reply:
sorted, client.CollectionDefinition is now a struct